### PR TITLE
Proposal harvest endpoint

### PR DESCRIPTION
**Pull request by dgarcia360**
**Note:** This is an implementation proposal, do not start adapting SDKs until this is not merged.

Related issue: https://github.com/nemtech/catapult-rest/issues/250

New endpoints:

* `/account/{accountId}/harvest`: Gets an array of blocks where the account is the signer
* `/account/{accountId}/beneficiary`: Gets an array of blocks where the account is the beneficiary

### CODE REVIEW

**Comment by Vektrat** [comment]
also, can we make it so that an object is returned instead of an array?

```
{
    blocks: []
]
```

**Comment by dgarcia360** [comment]
@Vektrat We can return an object, but note that these other endpoints are returning an array directly:

- /blocks/1/limit/10
- /block/1/transactions
- /account/C67F465087EF681824805B7E9FF3B2728A4EE847DE044DE5D9FA415F7660B08E/transactions

**Comment by Vektrat** [comment]
> @Vektrat We can return an object, but note that these other endpoints are returning an array directly:
> 
>     * /blocks/1/limit/10
> 
>     * /block/1/transactions
> 
>     * /account/C67F465087EF681824805B7E9FF3B2728A4EE847DE044DE5D9FA415F7660B08E/transactions

yeah i know and i understand, we are planning to migrate those at some point as well and if we keep on piling up, the ball gets bigger

**Comment by Vektrat** [comment]
in the end we will be providing this as a direct array, pending to change other very similar endpoints at once in a future change

**Comment by dgarcia360** [comment]
@Vektrat Ok, please approve if correct.

File: spec/parameters/id.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: id
+in: query
+description: Identifier up to which elements are returned.
```
Comment (martinayora) [inline spec/parameters/id.yml]: Depending on the ordering parameter, elements with id smaller than given id will be returned, or elements with id greater than given id will be returned. Being the first option the default option.

Maybe it affects the "up to" definition for the id.

https://github.com/nemtech/catapult-rest/blob/master/rest/src/db/CatapultDb.js#L152-L159

**Comment by dgarcia360** [comment]
@martinayora @Vektrat I'm not too convinced about having ``id`` outside meta. As you know, the transaction-related endpoints include id inside ``meta``.

Also, with this latest changes could you confirm if de endpoint ``/blocks/1/limit/25`` response also includes the ``id`` field?

**Comment by dgarcia360** [comment]
Closed in favor of #93

**Pull request closed without merging**